File: 0047-permutations-ii/0047-permutations-ii.cpp
class Solution {
public:
    set<vector<int>> se;

    void traverse(vector<int> li , set<int> seen ,vector<int> nums){
        if ( li.size() == nums.size() ) {
            se.insert(li);
            return ;
        }
        for ( int i = 0 ; i < nums.size() ; i++){
            if ( seen.count(i) == 0){
                vector<int> ne = li;
                ne.push_back(nums[i]);
                set<int> s = seen;
                s.insert(i);
                traverse(ne , s ,nums);
            }
        }
    }
    vector<vector<int>> permuteUnique(vector<int>& nums) {
        vector<int> l ;
        set<int> seti;
        vector<vector<int>> ans;
        traverse(l , seti , nums);
        for(auto i : se){
            ans.push_back(i);
        }
        return ans;
    }
};

File: 0047-permutations-ii/0047-permutations-ii.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        ans = []
        n = len(nums)
        def traverse(li , seen ):
            nonlocal n , ans ,nums
            if len(li) == n :
                if li not in ans:
                    ans.append(li)
                return 
            for i in range(n):
                if i not in seen:
                    traverse(li+[nums[i]] , seen.union(set([i])))
        traverse([] ,set())
        return ans